src/main/scala/org/ergoplatform/ErgoLikeContext.scala

```diff
@@ -193,10 +202,28 @@ object ErgoLikeContext {
 
   implicit class ErgoBoxOps(val ebox: ErgoBox) extends AnyVal {
     def toTestBox(isCost: Boolean): Box = {
-      if (ebox == null) return null
       new CostingBox(isCost, ebox)
     }
   }
+
+  implicit val jsonEncoder: Encoder[ErgoLikeContext] = { ctx =>
+    Json.obj(
+      "currentHeight" -> ctx.currentHeight.asJson,
+      "lastBlockUtxoRoot" -> ctx.lastBlockUtxoRoot.asJson,
+      "minerPubkey" -> ctx.minerPubkey.asJson,
+      "headers" -> ctx.headers.toArray.toSeq.asJson,
+      "preHeader" -> ctx.preHeader.asJson,
+      "dataBoxes" -> ctx.dataBoxes.asJson,
```

Comment:
@aslesarenko Seems like `dataBoxes` can be converted to a property as well. Also while I'm at it `TODO currentHeight and minerPubkey should be calculated from PreHeader` from @catena2w can be implemented as well.

Reply:
@aslesarenko in ergo `Header` is a subclass of `PreHeader`. Should we do this as well?

Reply:
No, codegen will not be happy. This is not a big deal.